vsan/client.go

```diff
@@ -84,3 +88,72 @@ func (c *Client) VsanPerfQueryPerf(ctx context.Context, cluster *vimtypes.Manage
 	}
 	return res.Returnval, nil
 }
+
+// Creates the vsan object identities instance. This is to be queried from vsan health.
+var (
+	VsanQueryObjectIdentitiesInstance = vimtypes.ManagedObjectReference{
+		Type:  "VsanObjectSystem",
+		Value: "vsan-cluster-object-system",
+	}
+)
+
+// VsanQueryObjectIdentities return host uuid
+func (c *Client) VsanQueryObjectIdentities(ctx context.Context, cluster vimtypes.ManagedObjectReference) (*vsantypes.VsanObjectIdentityAndHealth, error) {
+	req := vsantypes.VsanQueryObjectIdentities{
+		This:    VsanQueryObjectIdentitiesInstance,
+		Cluster: cluster,
+	}
+
+	res, err := methods.VsanQueryObjectIdentities(ctx, c.serviceClient, &req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+
+}
+
+// QueryVsanObjects Queries the vsan object
+func (c *Client) QueryVsanObjects(ctx context.Context, uuids []string) (string, error) {
+
+	url := os.Getenv("VC_URL") // example: export VC_URL='https://username:password@vc-ip/sdk'
```

Comment:
I don't think we want to use env vars with the `Client` here, which already has a URL configured you can access using `c.vim25Client.URL()`

Reply:
Sure, Now I have passed it as param